json-rpc/src/methods.rs

```diff
@@ -64,6 +67,28 @@ impl JsonRpcService {
     pub fn chain_id(&self) -> ChainId {
         self.chain_id
     }
+
+    pub fn validate_batch_size_limit(&self, size: usize) -> Result<(), JsonRpcError> {
+        self.validate_size_limit("batch size", self.batch_size_limit, size)
+    }
+
+    pub fn validate_page_size_limit(&self, size: usize) -> Result<(), JsonRpcError> {
+        self.validate_size_limit("page size", self.page_size_limit, size)
```

Comment:
nit: should we have enums to represent the different size limits?

Reply:
and my preference for the size limit names is snake_case, but your call

Reply:
It's a message to client (in the struct field "name"). So keep it simple and human readable format.